src/main/java/com/example/servlet20/SearchServlet.java

```diff
@@ -1,35 +1,49 @@
 package com.example.servlet20;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import jakarta.servlet.ServletException;
 import jakarta.servlet.annotation.WebServlet;
 import jakarta.servlet.http.HttpServlet;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
+
 import java.io.IOException;
 import java.sql.*;
 import java.util.ArrayList;
 import java.util.List;
 
-@WebServlet("/cars/search")
+@WebServlet("/secure/cars/search")
 public class SearchServlet extends HttpServlet {
-    List<String> carNames = new ArrayList<>();
     @Override
-    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        final String message = req.getReader().readLine();
-        boolean found = false;
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        if (request.getInputStream() == null ) {
+            response.sendError(400, "Bad Request: The Json is empty!");
+            response.getWriter().println("You must enter the request details in a json format!");
+        }
+        JsonNode jsonNode = null;
+        while(request.getInputStream().available() > 0) {
+            jsonNode = objectMapper.readTree(request.getInputStream());
+        }
+        assert jsonNode != null;
+        String name = jsonNode.get("name").asText();
```

Comment:
the best practice would be creating a Pojo class instead of accessing a field in json